### PR TITLE
[Merged by Bors] - feat: port Order.SymmDiff

### PR DESCRIPTION
**Pull request by dupuisf**
mathlib3 SHA: 6eb334bd8f3433d5b08ba156b8ec3e6af47e1904

### CODE REVIEW

File: Mathlib/Order/SymmDiff.lean [hunk 1]
```diff
@@ -61,24 +62,25 @@ def symmDiff [HasSup α] [SDiff α] (a b : α) : α :=
 
 /-- The Heyting bi-implication is `(b ⇨ a) ⊓ (a ⇨ b)`. This generalizes equivalence of
 propositions. -/
-def bihimp [HasInf α] [HasHimp α] (a b : α) : α :=
+def bihimp [HasInf α] [HImp α] (a b : α) : α :=
   (b ⇨ a) ⊓ (a ⇨ b)
 #align bihimp bihimp
 
 -- mathport name: «expr ∆ »
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:69]:
```suggestion
```

File: Mathlib/Order/SymmDiff.lean [hunk 2]
```diff
@@ -61,24 +62,25 @@ def symmDiff [HasSup α] [SDiff α] (a b : α) : α :=
...
-infixl:100
-  " ∆ " =>/- This notation might conflict with the Laplacian once we have it. Feel free to put it in locale
+/- This notation might conflict with the Laplacian once we have it. Feel free to put it in locale
   `order` or `symm_diff` if that happens. -/
-  symmDiff
+/-- Notation for symmDiff -/
+infixl:100 " ∆ " =>  symmDiff
 
 -- mathport name: «expr ⇔ »
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:75]:
```suggestion
```

File: Mathlib/Order/SymmDiff.lean [hunk 3]
```diff
@@ -61,24 +62,25 @@ def symmDiff [HasSup α] [SDiff α] (a b : α) : α :=
...
+/-- Notation for bihimp -/
 infixl:100 " ⇔ " => bihimp
 
 theorem symm_diff_def [HasSup α] [SDiff α] (a b : α) : a ∆ b = a \ b ⊔ b \ a :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:79]: I suspect all `symm_diff`s in lemmas should be `symmDiff`

File: Mathlib/Order/SymmDiff.lean [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+#align bool.symm_diff_eq_bxor Bool.symm_diff_eq_bxor
+
+section GeneralizedCoheytingAlgebra
+
+variable [GeneralizedCoheytingAlgebra α] (a b c d : α)
+
+@[simp]
+theorem to_dual_symm_diff : toDual (a ∆ b) = toDual a ⇔ toDual b :=
+  rfl
+#align to_dual_symm_diff to_dual_symm_diff
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:107]:
```suggestion
theorem toDual_symmDiff : toDual (a ∆ b) = toDual a ⇔ toDual b :=
  rfl
#align to_dual_symm_diff toDual_symmDiff
```

File: Mathlib/Order/SymmDiff.lean [hunk 5]
```diff
@@ -0,0 +1,874 @@
...
+
+@[simp]
+theorem of_dual_bihimp (a b : αᵒᵈ) : ofDual (a ⇔ b) = ofDual a ∆ ofDual b :=
+  rfl
+#align of_dual_bihimp of_dual_bihimp
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:112]:
```suggestion
theorem ofDual_bihimp (a b : αᵒᵈ) : ofDual (a ⇔ b) = ofDual a ∆ ofDual b :=
  rfl
#align of_dual_bihimp ofDual_bihimp
```

File: Mathlib/Order/SymmDiff.lean [hunk 6]
```diff
@@ -0,0 +1,874 @@
...
+
+@[simp]
+theorem of_dual_symm_diff (a b : αᵒᵈ) : ofDual (a ∆ b) = ofDual a ⇔ ofDual b :=
+  rfl
+#align of_dual_symm_diff of_dual_symm_diff
+
+theorem bihimp_comm : a ⇔ b = b ⇔ a := by simp only [(· ⇔ ·), inf_comm]
+#align bihimp_comm bihimp_comm
+
+instance bihimp_is_comm : IsCommutative α (· ⇔ ·) :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SymmDiff.lean:240]: If you think the instance should be named:
```suggestion
instance bihimp_isCommutative : IsCommutative α (· ⇔ ·) :=
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
Looks good, except that you'll need to change nearly every theorem name

File: Mathlib/Order/SymmDiff.lean [hunk 7]
```diff
@@ -0,0 +1,872 @@
...
+/-!
+# Symmetric difference and bi-implication
+
+This file defines the symmetric difference and bi-implication operators in (co-)Heyting algebras.
+
+## Examples
+
+Some examples are
+* The symmetric difference of two sets is the set of elements that are in either but not both.
+* The symmetric difference on propositions is `xor`.
```
Comment (kbuzzard) [inline Mathlib/Order/SymmDiff.lean:19]: No it's not, it's `Xor'`.

File: Mathlib/Order/SymmDiff.lean [hunk 8]
```diff
@@ -0,0 +1,872 @@
...
+* The symmetric difference on `Bool` is `bxor`.
```
Comment (kbuzzard) [inline Mathlib/Order/SymmDiff.lean:20]: No it's not, it's `Bool.xor`.

File: Mathlib/Order/SymmDiff.lean [hunk 9]
```diff
@@ -0,0 +1,872 @@
...
+  rfl
+#align symm_diff_eq_xor symmDiff_eq_xor
+
+@[simp]
+theorem bihimp_iff_iff {p q : Prop} : p ⇔ q ↔ (p ↔ q) :=
+  (iff_iff_implies_and_implies _ _).symm.trans Iff.comm
+#align bihimp_iff_iff bihimp_iff_iff
+
+@[simp]
+theorem Bool.symmDiff_eq_bxor : ∀ p q : Bool, p ∆ q = xor p q := by decide
```
Comment (kbuzzard) [inline Mathlib/Order/SymmDiff.lean:95]: There should be no mention of `bxor` in this file -- it no longer exists. This theorem should be called `symmDiff_eq_xor`.

**Review by kbuzzard — APPROVED** [review]
This looks good to me. Thanks! I've left some comments.

**Comment by dupuisf** [comment]
Thanks! I think I got them all now...

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**